encfs/BlockFileIO.cpp

```diff
@@ -251,8 +260,8 @@ bool BlockFileIO::write(const IORequest &req) {
     }
 
     // Finally, write the damn thing!
-    if (!cacheWriteOneBlock(blockReq)) {
-      ok = false;
+    res = cacheWriteOneBlock(blockReq);
+    if (res) {
```

Comment:
I just committed a comment to help : write methods return  the very first result returned by `RawFileIO::write` so :
 0 in case of success, or -errno in case of failure.